### ISSUE TITLE
./rdm: error while loading shared libraries: libQt5Charts.so.5

### ISSUE THREAD

**Issue by lazyou** [redis/RedisDesktopManager#3924]
RDM Version: 0.9
Environment (OS name and version): ubuntu 16.04 ( x86_64)
Redis Server Version: 3.0.6

Steps to reproduce:

1. after build from source

2. 
```
/usr/share/redis-desktop-manager/bin                                            
➤ ls
qt.backup  qt.conf  rdm*  rdm.png  rdm.sh

/usr/share/redis-desktop-manager/bin                                            
➤ ./rdm 

```

Expected result: open RDM

Actual Result: ./rdm: error while loading shared libraries: libQt5Charts.so.5: cannot open shared object file: No such file or directory

**Comment by uglide** [comment]
You need to install qt59charts-no-lgpl

**Comment by iselcuk** [comment]
@uglide ![image](https://user-images.githubusercontent.com/11681283/32959030-52e7c682-cbd1-11e7-8697-2f300055a7d6.png)

**Comment by colegatron** [comment]
Solved!

When you build from source the libraries are installed (at least in ubuntu) in /opt/qt59, so the easiest way of workaround it is running "source /opt/qt59/bin/qt59-env.sh" prior to run rdm.

So just update your /usr/share/redis-desktop-manager/bin/rdm.sh:

```
#!/bin/bash

source /opt/qt59/bin/qt59-env.sh
DIR=$(dirname "$(readlink -f "$0")")
export LD_LIBRARY_PATH="$DIR/../lib":$LD_LIBRARY_PATH
$DIR/rdm
```

**Comment by artamonovdev** [comment]
This application failed to start because it could not find or load the Qt platform plugin "xcb"

**Comment by teneon** [comment]
I am trying to install RedisDesktopManager from source for the past 3 hours. I have searched everywhere and nothing really helps. Even adding "source /opt/qt59/bin/qt59-env.sh" as suggested above.

I always end up with the same error when i try to run
/usr/share/redis-desktop-manager/bin$ ./rdm 
./rdm: error while loading shared libraries: libQt5Charts.so.5: cannot open shared object file: No such file or directory

I even tried to reinstall QT, i've checked if Charts are selected in the Maintaince tool for QT and they are. Any ideas :)?

best regards!

**Comment by uglide** [comment]
@teneon you have mixed 2 Qt installations from ppa and from official installer (/opt/qt59/ is ppa version)
Try to remove Qt installer version and re-install qt from ppa

**Comment by denisgrib** [comment]
Even adding "source /opt/qt59/bin/qt59-env.sh" as suggested above.

```
$ sudo ./rdm.sh                                                                                                                                                                                          
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
App font: "Open Sans" 11
("en")
Detected locale: "en"
New config already exist. Ignore old configs.
Permission changed
QQmlApplicationEngine failed to load component
qrc:///app.qml:66 Type ConnectionSettignsDialog unavailable
qrc:///ConnectionSettignsDialog.qml:458 Type ImageButton unavailable
qrc:///common/ImageButton.qml:2 module "QtQuick.Controls" version 2.2 is not installed

Rendering backend: ""
^C

```

**Comment by denisgrib** [comment]
" Qt installations from ppa and from official installer ..."
```
Reading package lists... Done
W: The repository 'http://ppa.launchpad.net/canonical-qt5-edgers/ubuntu1204-qt5/ubuntu xenial Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: Failed to fetch http://ppa.launchpad.net/canonical-qt5-edgers/ubuntu1204-qt5/ubuntu/dists/xenial/main/binary-amd64/Packages  404  Not Found
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by uglide** [comment]
Hm, do you use custom Qt ppa?

**Comment by denisgrib** [comment]
```
Ign:12 http://ppa.launchpad.net/canonical-qt5-edgers/ubuntu1204-qt5/ubuntu xenial InRelease                                                                     
Hit:13 http://ppa.launchpad.net/gophers/archive/ubuntu xenial InRelease                                                             
Hit:14 http://repo.percona.com/apt xenial InRelease                                                
Hit:16 http://ppa.launchpad.net/nilarimogard/webupd8/ubuntu xenial InRelease                       
Get:17 http://ppa.launchpad.net/ondrej/php/ubuntu xenial InRelease [23.9 kB]   
Err:18 http://ppa.launchpad.net/canonical-qt5-edgers/ubuntu1204-qt5/ubuntu xenial Release
  404  Not Found
Get:19 http://ppa.launchpad.net/ondrej/php/ubuntu xenial/main amd64 Packages [45.9 kB]
Get:20 http://ppa.launchpad.net/ondrej/php/ubuntu xenial/main i386 Packages [45.7 kB]
Reading package lists... Done                      
E: The repository 'http://ppa.launchpad.net/canonical-qt5-edgers/ubuntu1204-qt5/ubuntu xenial Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```
this is not official ppa?

**Comment by uglide** [comment]
@spaceblock RDM uses  Stephan Binner Qt ppa https://github.com/uglide/RedisDesktopManager/blob/0.9/src/configure#L66

**Comment by denisgrib** [comment]
@uglide ok,
```
This application failed to start because it could not find or load the Qt platform plugin "xcb"
in "".

Reinstalling the application may fix this problem.
Error: Can't launch CrashReporter!
[1]    5295 abort (core dumped)  rdm

```

**Comment by Sentoki** [comment]
I have all same problems, does anybody have solution for ubuntu 16.04?

**Comment by uglide** [comment]
@spaceblock @Sentoki 
Good starting point is 
`ldd rdm`
Copy paste output here, I will be easier to understand what is going on in your system

**Comment by denisgrib** [comment]
@uglide 
```
> $ ldd /usr/bin/rdm                                                                                                                                                                                       
/usr/bin/rdm: /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5: version `Qt_5' not found (required by /usr/bin/rdm)
/usr/bin/rdm: /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5: version `Qt_5' not found (required by /usr/bin/rdm)
/usr/bin/rdm: /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5: version `Qt_5' not found (required by /usr/bin/rdm)
/usr/bin/rdm: /usr/lib/x86_64-linux-gnu/libQt5Quick.so.5: version `Qt_5' not found (required by /usr/bin/rdm)
/usr/bin/rdm: /usr/lib/x86_64-linux-gnu/libQt5Network.so.5: version `Qt_5' not found (required by /usr/bin/rdm)
/usr/bin/rdm: /usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.9' not found (required by /usr/bin/rdm)
/usr/bin/rdm: /usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5' not found (required by /usr/bin/rdm)
	linux-vdso.so.1 =>  (0x00007fff2f4c6000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f72781da000)
	libcrypto.so.1.1 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1 (0x00007f7277d5f000)
	libQt5Charts.so.5 => not found
	libQt5Widgets.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5 (0x00007f72776d2000)
	libQt5Quick.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Quick.so.5 (0x00007f7277302000)
	libQt5Gui.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5 (0x00007f7276dba000)
	libQt5Qml.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5 (0x00007f727698e000)
	libQt5Network.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Network.so.5 (0x00007f727849f000)
	libQt5Core.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Core.so.5 (0x00007f72764b8000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f727629b000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f7275f19000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f7275d03000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f7275939000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f7275735000)
	libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007f72754e2000)
	libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f72751d1000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f7274e97000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f7274b8e000)
	libpng12.so.0 => /lib/x86_64-linux-gnu/libpng12.so.0 (0x00007f7274969000)
	libharfbuzz.so.0 => /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007f727470b000)
	libGL.so.1 => /usr/lib/x86_64-linux-gnu/mesa/libGL.so.1 (0x00007f7274498000)
	libdouble-conversion.so.1 => /usr/lib/x86_64-linux-gnu/libdouble-conversion.so.1 (0x00007f7274287000)
	libproxy.so.1 => /usr/lib/x86_64-linux-gnu/libproxy.so.1 (0x00007f7274066000)
	libicui18n.so.55 => /usr/lib/x86_64-linux-gnu/libicui18n.so.55 (0x00007f7273c04000)
	libicuuc.so.55 => /usr/lib/x86_64-linux-gnu/libicuuc.so.55 (0x00007f7273870000)
	libpcre16.so.3 => /usr/lib/x86_64-linux-gnu/libpcre16.so.3 (0x00007f7273608000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f7273400000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f72783f4000)
	libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6 (0x00007f72731f8000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f7272f87000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f7272d65000)
	libfreetype.so.6 => /usr/lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007f7272abb000)
	libgraphite2.so.3 => /usr/lib/x86_64-linux-gnu/libgraphite2.so.3 (0x00007f7272895000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f727266c000)
	libxcb-dri3.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri3.so.0 (0x00007f7272469000)
	libxcb-present.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-present.so.0 (0x00007f7272266000)
	libxcb-sync.so.1 => /usr/lib/x86_64-linux-gnu/libxcb-sync.so.1 (0x00007f727205f000)
	libxshmfence.so.1 => /usr/lib/x86_64-linux-gnu/libxshmfence.so.1 (0x00007f7271e5c000)
	libglapi.so.0 => /usr/lib/x86_64-linux-gnu/libglapi.so.0 (0x00007f7271c2c000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007f7271a1a000)
	libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007f7271817000)
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007f7271611000)
	libX11-xcb.so.1 => /usr/lib/x86_64-linux-gnu/libX11-xcb.so.1 (0x00007f727140f000)
	libxcb-glx.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-glx.so.0 (0x00007f72711f6000)
	libxcb-dri2.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri2.so.0 (0x00007f7270ff1000)
	libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007f7270deb000)
	libdrm.so.2 => /usr/lib/x86_64-linux-gnu/libdrm.so.2 (0x00007f7270bda000)
	libicudata.so.55 => /usr/lib/x86_64-linux-gnu/libicudata.so.55 (0x00007f726f123000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f726ef1f000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f726ed19000)
```

**Comment by denisgrib** [comment]
```
/usr/lib/x86_64-linux-gnu                                                                                                                                                 [13:14:09] 
> $ ll libQt5*so.5                                                                                                                                                                                         
lrwxrwxrwx 1 root root 19 May 12  2017 libQt5Core.so.5 -> libQt5Core.so.5.5.1
lrwxrwxrwx 1 root root 19 May 12  2017 libQt5DBus.so.5 -> libQt5DBus.so.5.5.1
lrwxrwxrwx 1 root root 35 May 12  2017 libQt5EglDeviceIntegration.so.5 -> libQt5EglDeviceIntegration.so.5.5.1
lrwxrwxrwx 1 root root 23 Aug 24  2017 libQt5Feedback.so.5 -> libQt5Feedback.so.5.0.0
lrwxrwxrwx 1 root root 18 May 12  2017 libQt5Gui.so.5 -> libQt5Gui.so.5.5.1
lrwxrwxrwx 1 root root 25 Aug 24  2017 libQt5Multimedia.so.5 -> libQt5Multimedia.so.5.5.1
lrwxrwxrwx 1 root root 22 May 12  2017 libQt5Network.so.5 -> libQt5Network.so.5.5.1
lrwxrwxrwx 1 root root 21 May 12  2017 libQt5OpenGL.so.5 -> libQt5OpenGL.so.5.5.1
lrwxrwxrwx 1 root root 24 Aug 24  2017 libQt5Organizer.so.5 -> libQt5Organizer.so.5.0.0
lrwxrwxrwx 1 root root 26 Aug 24  2017 libQt5Positioning.so.5 -> libQt5Positioning.so.5.5.1
lrwxrwxrwx 1 root root 27 May 12  2017 libQt5PrintSupport.so.5 -> libQt5PrintSupport.so.5.5.1
lrwxrwxrwx 1 root root 18 Aug 24  2017 libQt5Qml.so.5 -> libQt5Qml.so.5.5.1
lrwxrwxrwx 1 root root 20 Aug 24  2017 libQt5Quick.so.5 -> libQt5Quick.so.5.5.1
lrwxrwxrwx 1 root root 24 Aug 24  2017 libQt5QuickTest.so.5 -> libQt5QuickTest.so.5.5.1
lrwxrwxrwx 1 root root 18 May 12  2017 libQt5Sql.so.5 -> libQt5Sql.so.5.5.1
lrwxrwxrwx 1 root root 18 Aug 24  2017 libQt5Svg.so.5 -> libQt5Svg.so.5.5.1
lrwxrwxrwx 1 root root 19 May 12  2017 libQt5Test.so.5 -> libQt5Test.so.5.5.1
lrwxrwxrwx 1 root root 21 Aug 24  2017 libQt5WebKit.so.5 -> libQt5WebKit.so.5.5.1
lrwxrwxrwx 1 root root 28 Aug 24  2017 libQt5WebKitWidgets.so.5 -> libQt5WebKitWidgets.so.5.5.1
lrwxrwxrwx 1 root root 22 May 12  2017 libQt5Widgets.so.5 -> libQt5Widgets.so.5.5.1
lrwxrwxrwx 1 root root 24 Nov 30  2015 libQt5X11Extras.so.5 -> libQt5X11Extras.so.5.5.1
lrwxrwxrwx 1 root root 21 May 12  2017 libQt5XcbQpa.so.5 -> libQt5XcbQpa.so.5.5.1
lrwxrwxrwx 1 root root 18 May 12  2017 libQt5Xml.so.5 -> libQt5Xml.so.5.5.1
```

**Comment by uglide** [comment]
@spaceblock You need Qt 5.9.1 + rdm.sh update https://github.com/uglide/RedisDesktopManager/issues/3924#issuecomment-345652988

**Comment by denisgrib** [comment]
@uglide 
-- I install last version from Stephan Binner Qt ppa
-- I have 
```
> $ ll /opt/qt59/bin/qt59-env.sh                                                                                                                                                                           
-rwxr-xr-x 1 root root 497 Apr  3  2017 /opt/qt59/bin/qt59-env.sh
```
-- I add ```source /opt/qt59/bin/qt59-env.sh``` to rdm.sh

**finally:**
```
> $ ./rdm.sh                                                                                                                                                                                               
App font: "Open Sans" 11
("en")
Detected locale: "en"
New config already exist. Ignore old configs.
Permission changed
QQmlApplicationEngine failed to load component
qrc:///app.qml:66 Type ConnectionSettignsDialog unavailable
qrc:///ConnectionSettignsDialog.qml:458 Type ImageButton unavailable
qrc:///common/ImageButton.qml:2 module "QtQuick.Controls" version 2.2 is not installed

Rendering backend: ""
```
```
> $ sudo ./rdm.sh                                                                                                                                                                                          
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
App font: "Open Sans" 11
("en")
Detected locale: "en"
New config already exist. Ignore old configs.
Permission changed
QQmlApplicationEngine failed to load component
qrc:///app.qml:66 Type ConnectionSettignsDialog unavailable
qrc:///ConnectionSettignsDialog.qml:458 Type ImageButton unavailable
qrc:///common/ImageButton.qml:2 module "QtQuick.Controls" version 2.2 is not installed

Rendering backend: ""
```

**Comment by uglide** [comment]
@spaceblock It's bug. Please rebuild with latest 0.9 commit https://github.com/uglide/RedisDesktopManager/commit/0ebbfd88683cc12637282063ff4f9b8e47a8d29f

**Comment by JonyD** [comment]
(Before I had successfully installed the v 0.9.0-develop). Today I wanted to update to the latest version.
I have ubuntu 16.04 + qt59 and checked out the latest version (to date) v0.9 bc50ec346b19c9 / 2018-05-08 17:52 Igor Malinovskiy.
It got the same error message when runnign ./rdm:

> ./rdm: error while loading shared libraries: libQt5Charts.so.5: cannot open shared object file: No such file or directory

To solve the problem I had to do:
- sudo add-apt-repository --yes ppa:beineri/opt-qt59-xenial
- sudo apt-get update -y
- sudo apt-get install qt59base qt59imageformats qt59tools qt59declarative qt59quickcontrols qt59quickcontrols2 qt59charts-no-lgpl qt59svg -y
- (apparently 1 lib was missing (qt59quickcontrols2 I think ) and was not being installed in configure, even when running sudo ./configure )
- and then run the ./configure again + all the installation steps described in http://docs.redisdesktop.com/en/latest/install/
